### PR TITLE
Port MYFACES-4468  (f:selectItemGroup)

### LINKED ISSUE #22790: Port MYFACES-4468 to Faces 4.0
`https://issues.apache.org/jira/projects/MYFACES/issues/MYFACES-4468`

Also tried to address TCK signature failure:

```
[javatest.batch] Added Fields
[javatest.batch] ------------
[javatest.batch]
[javatest.batch] jakarta.faces.component.UISelectItemGroups: field public final static java.lang.String jakarta.faces.component.UISelectItemGroups.COMPONENT_FAMILY = "jakarta.faces.SelectItemGroups"
```

### PR DESCRIPTION
**Pull request by volosied**
fixes #22790
for Spec Issue #20838 (test also added in this PR) and test also added for #20840 (selectItemGroups)

### CODE REVIEW

**Comment by volosied** [comment]
#build

**Comment by volosied** [comment]
#build

File: dev/io.openliberty.org.apache.myfaces.4.0_fat/fat/src/io/openliberty/org/apache/myfaces40/fat/tests/SelectItemTests.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.websphere.simplicity.log.Log;
+
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.HttpUtils;
+
+/**
+ *  Verifies that the f:selectItemGroup and f:selectItemGroup output the correct selections.
```
Comment (pnicolucci) [inline dev/io.openliberty.org.apache.myfaces.4.0_fat/fat/src/io/openliberty/org/apache/myfaces40/fat/tests/SelectItemTests.java]: f:selectItemGroup and f:selectItemGroup**s**

File: dev/io.openliberty.org.apache.myfaces.4.0_fat/fat/src/io/openliberty/org/apache/myfaces40/fat/tests/SelectItemTests.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    @Server("faces40_selectItemsServer")
+    public static LibertyServer server;
+
+    @Rule
+    public TestName name = new TestName();
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        ShrinkHelper.defaultDropinApp(server, APP_NAME + ".war", 
+            "io.openliberty.org.apache.faces40.fat.beans.selectitemgroup",
```
Comment (pnicolucci) [inline dev/io.openliberty.org.apache.myfaces.4.0_fat/fat/src/io/openliberty/org/apache/myfaces40/fat/tests/SelectItemTests.java]: How would you feel about changing the order of the package a bit `io.openliberty.org.apache.faces40.fat.selectitemgroup.beans`

File: dev/io.openliberty.org.apache.myfaces.4.0_fat/test-applications/SelectItemTests.war/resources/selectItemGroups.xhtml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    <h:selectOneMenu id="menu">
+        <f:selectItemGroups value="#{selectItemGroupsBean.destinationList}" var="destination" itemValue="#{destination.region}" itemLabel="#{destination.region}">
+            <f:selectItems value="#{destination.countries}" var="country" itemValue="#{country.id}" itemLabel="#{country.name}" />
+        </f:selectItemGroups>
+    </h:selectOneMenu>
+ </h:form>
+ 
+ </body>
+ </html>
+
```
Comment (pnicolucci) [inline dev/io.openliberty.org.apache.myfaces.4.0_fat/test-applications/SelectItemTests.war/resources/selectItemGroups.xhtml]: new line

**Comment by volosied** [comment]
#build

**Review by pnicolucci — APPROVED** [review]

**Comment by pnicolucci** [comment]
#libby

**Pull request merged**